test/e2e/instrumentation/monitoring/custom_metrics_stackdriver.go

```diff
@@ -160,6 +255,29 @@ func testAdapter(f *framework.Framework, kubeClient clientset.Interface, customM
 	}
 }
 
+func verifyResponseFromExternalMetricsAPI(f *framework.Framework, externalMetricsClient externalclient.ExternalMetricsClient) {
+	req1, _ := labels.NewRequirement("resource.type", selection.Equals, []string{"k8s_pod"})
+	req2, _ := labels.NewRequirement("resource.label.pod_name", selection.Exists, []string{})
+	req3, _ := labels.NewRequirement("resource.label.location", selection.NotEquals, []string{"dummy"})
+	req4, _ := labels.NewRequirement("resource.label.namespace_name", selection.In, []string{string(f.Namespace.Name), "dummy"})
+	req5, _ := labels.NewRequirement("resource.label.cluster_name", selection.NotIn, []string{"foo", "bar"})
+	values, err := externalMetricsClient.
+		NamespacedMetrics("dummy").
+		List("custom.googleapis.com|"+CustomMetricName, labels.NewSelector().Add(*req1, *req2, *req3, *req4, *req5))
+	if err != nil {
+		framework.Failf("Failed query: %s", err)
+	}
+	if len(values.Items) != 1 {
```

Comment:
What if more than a single test runs in the same project (I think it's possible with current e2e job setup)? Maybe we should add a filter using real namespace name?

Reply:
We filter for the real namespace, but using In operator. Added a comment that it serves a purpose of avoiding collisions.

Reply:
I didn't realize 'dummy' was the real namespace name. Why not use the e2e namespace (which, I think, is unique)? However, it's not very important, so feel free to ignore this comment if it's too much work or you disagree.

Reply:
In the actual filter, I use the real e2e namespace name: string(f.Namespace.Name). 'dummy' is not a real namespace name, in fact I expect there won't be such namespace. I intentionally use it as a namespace path argument to emphasize it doesn't matter.

I think any changes here would be small and effortless, so I'm fine with applying whatever you suggest. I'm just not 100% sure what is your proposal. Do you think it needs some more comments to avoid confusion?

Reply:
Sorry, I missed that you actually pass 2 values with In, for some reason I thought you only pass dummy. Please disregard my comment, everything is fine.